polkadot/runtime/parachains/src/hrmp.rs

```diff
@@ -640,6 +659,132 @@ pub mod pallet {
 
 			Ok(Some(<T as Config>::WeightInfo::force_open_hrmp_channel(cancel_request)).into())
 		}
+
+		/// Establish an HRMP channel between two system chains. If the channel does not already
+		/// exist, the transaction fees will be refunded to the caller. The system does not take
+		/// deposits for channels between system chains, and automatically sets the message number
+		/// and size limits to the maximum allowed by the network's configuration.
+		///
+		/// Arguments:
+		///
+		/// - `sender`: A system chain, `ParaId`.
+		/// - `recipient`: A system chain, `ParaId`.
+		///
+		/// Any signed origin can call this function, but _both_ inputs MUST be system chains. If
+		/// the channel does not exist yet, there is no fee.
+		#[pallet::call_index(8)]
+		#[pallet::weight(<T as Config>::WeightInfo::establish_system_channel())]
+		pub fn establish_system_channel(
+			origin: OriginFor<T>,
+			sender: ParaId,
+			recipient: ParaId,
+		) -> DispatchResultWithPostInfo {
+			let _caller = ensure_signed(origin)?;
+
+			// to recognize current chains needs
+			// https://github.com/paritytech/polkadot-sdk/pull/1406
+			ensure!(
+				sender.is_system() && recipient.is_system(),
+				Error::<T>::ChannelCreationNotAuthorized
+			);
+
+			let config = <configuration::Pallet<T>>::config();
+			let max_message_size = config.hrmp_channel_max_message_size;
+			let max_capacity = config.hrmp_channel_max_capacity;
+
+			Self::init_open_channel(sender, recipient, max_capacity, max_message_size)?;
+			Self::accept_open_channel(recipient, sender)?;
+
+			Self::deposit_event(Event::HrmpSystemChannelOpened(
+				sender,
+				recipient,
+				max_capacity,
+				max_message_size,
+			));
+
+			Ok(Pays::No.into())
+		}
+
+		/// Update the deposits held for an HRMP channel. If at least one of the chains is a system
+		/// chain, any deposits held will be unreserved.
+		///
+		/// Arguments:
+		///
+		/// - `sender`: A chain, `ParaId`.
+		/// - `recipient`: A chain, `ParaId`.
+		///
+		/// Any signed origin can call this function.
+		#[pallet::call_index(9)]
+		#[pallet::weight(<T as Config>::WeightInfo::poke_channel_deposits())]
+		pub fn poke_channel_deposits(
```

Comment:
If this call "succeeds", as in either the sender or the recipient is a system chain AND the deposits do get updated, would we not want the caller to not pay any fees like in the other extrinsic (`Pays::No`)?

Reply:
I considered it, it would be fine. But if deposits go down, there is already enough incentive to update them.